_posts/2020-07-21-tips-and-tricks-for-running-strimzi-with-kubectl.md

```diff
@@ -0,0 +1,129 @@
+| Kafka Bridge          | kafkabridge       | kb            |
+| Kafka Rebalance       | kafkarebalance    | kr            |
+
+
+This is really handy because some of the resources have really long names. 
+So learning the short names can save a lot of time while operating Strimzi or working on new features.
+
+## Resource categories
+
+The custom resources can be also grouped into categories.
+The name of the category can be used in the `kubectl` commands instead of the individual resource types.
+All our custom resources are in the category `strimzi`.
+You can use it for example to get all the Strimzi resources with one command.
+For example something like `kubectl get strimzi` which lists all the different Strimzi resources which exist in given namespace.
+
+```
+$ kubectl get strimzi
+NAME                                DESIRED KAFKA REPLICAS   DESIRED ZK REPLICAS
+kafka.kafka.strimzi.io/my-cluster   3                        3
+
+NAME                                          PARTITIONS   REPLICATION FACTOR
+kafkatopic.kafka.strimzi.io/kafka-test-apps   3            3
+
+NAME                                 AUTHENTICATION   AUTHORIZATION
+kafkauser.kafka.strimzi.io/my-user   tls              simple
+```
+
+You can also combine this with other commands.
+For example while developing or testing new Strimzi features, I quite often need to delete all Strimzi resources.
+Doing it resource by resource would take a long time.
+But I can do it all in a single command:
+
+```
+$ kubectl delete $(kubectl get strimzi -o name)
+kafka.kafka.strimzi.io "my-cluster" deleted
+kafkatopic.kafka.strimzi.io "kafka-test-apps" deleted
+kafkauser.kafka.strimzi.io "my-user" deleted
+```
+
+The inner command - `kubectl get strimzi -o name` - will return the types and the resource names.
+This is enabled by the `-o name` option.
+
+```
+$ kubectl get strimzi -o name
+kafka.kafka.strimzi.io/my-cluster
+kafkatopic.kafka.strimzi.io/kafka-test-apps
+kafkauser.kafka.strimzi.io/my-user
+```
+
+And when this is passed into the `kubectl delete` command, it will delete all Strimzi resources.
+
+## Querying the status subresources
+
+We already saw how the `-o name` option can be used to get the output in the `type/name` format.
+There are also other values you can pass to the `-o` option.
+For example by using `-o yaml` you can get the output in YAML format.
+Or `-o json` will return it as JSON.
+You can see all the options in `kubectl get --help` or in the [`kubectl` docs](https://kubernetes.io/docs/reference/kubectl/overview/#formatting-output).
+
+One of the most useful options is `jsonpath`.
+It allows you to pass [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) expression which will be evaluated while querying the Kubernetes API.
+The JSONPath expression can be used to extract specific parts of the custom resource.
+For example, you can easily get the bootstrap address from the status of the Kafka custom resource and use it in your Kafka clients.
```

Comment:
```suggestion
For example, you can get the bootstrap address from the status of the Kafka custom resource and use it in your Kafka clients.
```
This is going to sound so petty, but I think JSONPath is not easy, constructing new paths is always a hassle I find, especially when working with arrays, maybe remove easily? :P